patches/com/mojang/blaze3d/pipeline/RenderTarget.java.patch

```diff
@@ -0,0 +1,134 @@
+--- a/com/mojang/blaze3d/pipeline/RenderTarget.java
++++ b/com/mojang/blaze3d/pipeline/RenderTarget.java
+@@ -25,17 +_,31 @@
+     public int viewWidth;
+     public int viewHeight;
+     public final boolean useDepth;
++    public final boolean useStencil;
+     public int frameBufferId;
+     protected int colorTextureId;
+     protected int depthBufferId;
++    protected int stencilBufferId;
+     private final float[] clearChannels = Util.make(() -> new float[]{1.0F, 1.0F, 1.0F, 0.0F});
+     public int filterMode;
+ 
+     public RenderTarget(boolean p_166199_) {
+-        this.useDepth = p_166199_;
++        this(p_166199_, false);
++    }
++
++    public RenderTarget(boolean useDepth, boolean useStencil) {
++        this.useDepth = useDepth;
++        this.useStencil = useStencil;
+         this.frameBufferId = -1;
+         this.colorTextureId = -1;
+         this.depthBufferId = -1;
++
++        if (!useDepth && useStencil) {
++            var capabilities = org.lwjgl.opengl.GL.getCapabilities();
++            if (!capabilities.GL_ARB_texture_stencil8 && !capabilities.OpenGL44) {
++                throw new UnsupportedOperationException("Stencil-only buffers require GL_ARB_texture_stencil8 OR OpenGL 4.4");
++            }
++        }
```

Comment:
OpenGL technically supports it, and it's a supported combination of parameters, so rather than getting a confusing error about framebuffer completeness with no exception, I'd rather throw an exception beforehand.

You are correct though, in that the average user would probably not hit this branch

Reply:
If this is perf-critical code I would suggest dropping this check.  Presumably the modder raising the incorrect flag combo would see a breakage elsewhere.

If not, we can keep it for informative purposes.